include/iot_socket.h

```diff
@@ -188,6 +189,7 @@ extern int32_t iotSocketRecv (int32_t socket, void *buf, uint32_t len);
 
 /**
   \brief         Receive data on a socket.
+  \note          Function can be used to poll the socket by setting len to 0. If data is available it will return 0.
```

Comment:
See previous comment.

Reply:
Paul, thanks a lot for spotting the issue - fully agree that the documentation is not complete and really appreciate your effort.

We must however view the complete documentation, not just the header file and I would suggest that we form a usability team that also reviews aspects such as VS Code appearance.  The result of that work should be guidelines (that are simple, but allow consistency). It could be a rework of https://github.com/ARM-software/CMSIS_5/blob/develop/CMSIS/DoxyGen/How2Doc.txt

In the specific case, the complete function documentation is here https://github.com/MDK-Packs/IoT_Socket/blob/develop/documentation/src/IoT_Socket_API.txt#L293 and the framed documentation (that is composed from header file + text file) is here
https://mdk-packs.github.io/IoT_Socket/html/group__iotSocketAPI.html#ga9b71bd827a43f2fed4331e8087915689

To solve the problem you have spotted, I suggest not to add another \note (as the description has already one), but change the brief description along with parameter description as shown below. As you see, I have used // to separate between return value and description - not sure if it is a better way, but the '=' is somewhat confusing for value = 0 and I found it inconsistent to describe the value in the front or at the end of the return value list.

Let me know what you think, I'm open to suggestions.

Reinhard
```c
/**
  \brief         Receive data or poll for data available on a connected socket.
  \param[in]     socket   socket identification number.
  \param[out]    buf      pointer to buffer where data should be stored.
  \param[in]     len      length of buffer (in bytes), set len=0 to poll for data available.
  \return        status information:
                 - value = 0                     // for len = 0 it indicates that data are available.
                 - value > 0                     // number of bytes received.
                 - \ref IOT_SOCKET_ESOCK         // Invalid socket.
                 - \ref IOT_SOCKET_EINVAL        // Invalid argument (pointer to buffer or length).
                 - \ref IOT_SOCKET_ENOTCONN      // Socket is not connected.
                 - \ref IOT_SOCKET_ECONNRESET    // Connection reset by the peer.
                 - \ref IOT_SOCKET_ECONNABORTED  // Connection aborted locally.
                 - \ref IOT_SOCKET_EAGAIN        // Operation would block or timed out (may be called again).
                 - \ref IOT_SOCKET_ERROR         // Unspecified error.
 */
extern int32_t iotSocketRecv (int32_t socket, void *buf, uint32_t len);
```

Reply:
I would also go with the suggested approach from Reinhard and extend brief descriptions, parameter description and return value. It should provide enough information when looking just at the header file and fits within the built documentation.

I would just slightly update the proposal (stick with '=' instead of '//' for return values) to be consistent with existing approach used in many places (not just IoT Socket).

How about the following:

```
/**
  \brief         Receive data or check if data is available on a connected socket.
  \param[in]     socket   socket identification number.
  \param[out]    buf      pointer to buffer where data should be stored.
  \param[in]     len      length of buffer (in bytes), set len = 0 to check if data is available.
  \return        status information:
                 - number of bytes received (>=0), if len != 0.
                 - 0                             = Data is available (len = 0).
                 - \ref IOT_SOCKET_ESOCK         = Invalid socket.
                 - \ref IOT_SOCKET_EINVAL        = Invalid argument (pointer to buffer or length).
                 - \ref IOT_SOCKET_ENOTCONN      = Socket is not connected.
                 - \ref IOT_SOCKET_ECONNRESET    = Connection reset by the peer.
                 - \ref IOT_SOCKET_ECONNABORTED  = Connection aborted locally.
                 - \ref IOT_SOCKET_EAGAIN        = Operation would block or timed out (may be called again).
                 - \ref IOT_SOCKET_ERROR         = Unspecified error.
 */
extern int32_t iotSocketRecv (int32_t socket, void *buf, uint32_t len);
```

If we agree on that I would then extend this also to iotSocketRecvFrom and iotSocketSend/iotSocketSendTo.